### PR TITLE
Codesplitting on all routes using loadable-components

### PR DESCRIPTION
**Pull request by LudvigHz**
Also adds some tree-shaking and webpack optimization tweaks.

~The vendors chunk is still quite large. With zxcvbn using \~700KiB and recharts using \~500KiB. We should find a way to split these out from the main vendor chunk, and load async. (looking at this now).~

In the mean time, this should be alot better.

**Before:**
```
[1] WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
[1] This can impact web performance.
[1] Assets: 
[1]   a10340ceb228c90cbeb6.ttf (8.95 MiB)
[1]   app.20320511.js (6.44 MiB)
[1]   c2bd9cd4b22ff17d91c4.jpg (280 KiB)
[1]   5c3185a787083623adb4.png (348 KiB)
[1]   fa60fac19a5deb240e84.png (819 KiB)
[1]   45b8e04fc7022ab64c11.png (331 KiB)
[1]   vendors.ef812859.js (5.78 MiB)
[1]   styles.c6b02716d66620b130e6.css (272 KiB)
[1]


[1] Entrypoints:
[1]   app (12.6 MiB)
[1]       vendors.ef812859.js
[1]       styles.6df2013d.js
[1]       styles.c6b02716d66620b130e6.css
[1]       app.20320511.js
```
image

**After:**
```
[1] WARNING in asset size limit: The following asset(s) exceed the recommended size limit (244 KiB).
[1] This can impact web performance.
[1] Assets: 
[1]   a10340ceb228c90cbeb6.ttf (8.95 MiB)
[1]   app.3f2c1a49.js (584 KiB)
[1]   c2bd9cd4b22ff17d91c4.jpg (280 KiB)
[1]   5c3185a787083623adb4.png (348 KiB)
[1]   45b8e04fc7022ab64c11.png (331 KiB)
[1]   fa60fac19a5deb240e84.png (819 KiB)
[1]   5677.168c437d.js (1.26 MiB)
[1]   8464.chunk.9dd1bb57.js (366 KiB)
[1]   1840.chunk.ec49e232.js (812 KiB)
[1]   styles.9c103041b67ea59185ad.css (272 KiB)

[1] Entrypoints:
[1]   app (2.16 MiB)
[1]       styles.85f811b8.js
[1]       styles.9c103041b67ea59185ad.css
[1]       5677.168c437d.js
[1]       app.3f2c1a49.js
```

image

### CODE REVIEW

File: server/pageRenderer.js [hunk 1]
```diff
@@ -43,25 +27,32 @@ function retrieveAssets() {
...
-  const { scripts, styles } = retrieveAssets();
-  const isSSR = body === '' ? 'false' : 'true';
+  const extractor = new ChunkExtractor({
+    statsFile: path.join(webpackClient.outputPath, 'loadable-stats.json'),
+    entrypoints: ['app'],
+  });
+  const collectedApp = extractor.collectChunks(app);
+
+  const { scripts, styles, links } = retrieveAssets(extractor);
+  const isSSR = app === undefined ? 'false' : 'true';
```
Comment (ivarnakken) [inline server/pageRenderer.js:47]: I don't understand why `isSSR` must be a string, care to explain? If not, perhaps do something like
```
const isSSR = app !== undefined;
```

**Review by ivarnakken — APPROVED** [review]
Woah this is some sexy cleanup 😍 Good job!!

File: server/pageRenderer.js (hunk 1 above)
Comment (LudvigHz) [reply server/pageRenderer.js:47]: If you see below, it's templated into the html. I guess it should work with a string :thinking:. But it doesn't really matter :shrug:

File: config/webpack.server.js [hunk 2]
```diff
@@ -48,7 +48,7 @@ module.exports = (env, argv) => {
       new FilterWarningsPlugin({
         // suppress conflicting order warnings from mini-css-extract-plugin.
         // see https://github.com/webpack-contrib/mini-css-extract-plugin/issues/250
-        exclude: /Conflicting order between:/,
+        exclude: /Conflicting order. Following module has been added/,
```
Comment (erlingfn) [inline config/webpack.server.js:51]: :see_no_evil: If we cannot see the warning, is there even a problem? :thinking:

Comment (LudvigHz) [reply config/webpack.server.js:51]: The warning doesn't really matter that much to as (we could fix it but eh :shrug:). And it's super annoying with all the spam every time you compile the app.

**Review by erlingfn — APPROVED** [review]
This looks great! Anything that simplifies server side rendering gets a thumbs up from me!

File: config/webpack.server.js (hunk 2 above)
Comment (erlingfn) [reply config/webpack.server.js:51]: Yeah. Even if we were to fix this, it would be better to do it in a separate PR.

**Comment by LudvigHz** [comment]
> simplifies server side rendering

Meanwhile, this PR adding more steps and making the SSR more complicated :upside_down_face:

**Comment by erlingfn** [comment]
Well it might make the process more complicated, but code looks simpler imo, and the library is well documented.

**Comment by erlingfn** [comment]
But whatever we think of the ssr, this is sorely needed.

**Pull request merged**